packaging/dependency/lxd.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package dependency
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/juju/packaging"
+)
+
+const blankSeries = ""
+
+// LXD is a dependency instance for installing lxd.
+var LXD packaging.Dependency = lxdDependency{}
+
+type lxdDependency struct{}
+
+// PackageList implements packaging.Dependency.
+func (lxdDependency) PackageList(series string) ([]packaging.Package, error) {
+	var pkg packaging.Package
+
+	switch series {
+	case "centos7", "opensuseleap":
+		return nil, errors.NotImplementedf("installing lxd on %s", series)
+	case "precice", "trusty", "xenial", "zesty", "artful",
```

Comment:
misspelled "precise" but we probably don't need it anyway.
we don't need zesty or artful either as those are definitely not support.d

Reply:
I will remove these.

Reply:
Should I move "precise" to L23 next to centos so we get a not-implemented error?

Reply:
Perhaps we need to add something like this for precise (and remove the [existing](https://github.com/juju/juju/blob/develop/container/lxd/initialisation_linux.go#L264-L266) check from ensureDependencies)

```go
case "precise":
     return errors.NotSupportedf("LXD containers on series %q", series)
```

Reply:
I have updated this to return NotSupported for centos/opensuseleap/precise. Let me know if that's fine or whether I should remove precise altogether.

Reply:
It seems to me that the pattern we are currently using is specify cases for exceptions and default for everything else. If we did it the other way round, we would need to patch this code for every new ubuntu release, right? While it would still break if you pass centos8 you should still be able to bootstrap something like eoan 

I *could* make the switch work with string.HasPrefix() to catch centosX stuff though...

Reply:
we do, but having that done as part of when we make a release that "now we support eoan" would be a reasonable time to do so. As you say, we haven't done it so far, so I don't think we have to right now. I was just contemplating what makes the most sense from a sanity check pov.